.buildkite/scripts/build_and_upload_packages.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+################################################################################
+# This file is responsible for building and uploading non-Docker packages, like
+# the Firecracker microVM kernel and the Firecracker rootfs.
+# It will not upload anything if a file of that name and SHA already exist
+# upstream.
+################################################################################
+
+source .buildkite/scripts/lib/artifacts.sh
+source .buildkite/scripts/lib/version.sh
+
+TAG="$(timestamp_and_sha_version)"
+
+(
+    echo "--- Building packages"
+    make dist/firecracker_kernel.tar.gz
+)
+
+# These will be uploaded to Cludsmith with just their basename.
+readonly PACKAGES=(
+    dist/firecracker_kernel.tar.gz
+)
+
+# This is where our images will ultimately be promoted to. It is the
+# registry we'll need to query to see if file with the same contents
+# already exists.
+readonly UPSTREAM_REGISTRY="${UPSTREAM_REGISTRY:-grapl/testing}"
+readonly UPLOAD_TO_REGISTRY="${UPLOAD_TO_REGISTRY:-grapl/raw}"
+
+# This is the list of packages that actually have different shas
+new_packages=()
+
+# Returns a string like `2de4f0c...`
+sha256_of_file() {
+    sha256sum "${1}" | awk '{print $1;}'
+}
+
+# Get the json query result of a named package.
+# Usage:
+#  query_package firecracker_kernel.tar.gz
+#  CHECKSUM=a1b2c3 query_package firecracker_kernel.tar.gz
+cloudsmith_query_package() {
+    queries=(
+        --query "name:${1}"
+    )
+    if [ -n "${CHECKSUM}" ]; then
+        queries+=(
+            --query "checksum:${CHECKSUM}"
+        )
```

Comment:
Great question!!!
The package name is required; the checksum is not.
I could see a world in which we'd change this function so that want to supply some other filters instead - in which case the $2 solution falls apart.

this is trying to present an interface like
```
def get_package(
  name,
  checksum=None,
  version=None,
  some_other_filters=None...
) -> 
```

the other approach would be to feed in an array of `--query` params, which is not the worst thing in the world

Reply:
Changed it to just feed in `--query=`